packages/plugin-core/src/commands/ShowPreview.ts

```diff
@@ -216,6 +217,75 @@ const handleAssetLink = async ({
   await ShowPreviewAssetOpener.openWithDefaultApp(assetFullPath);
 };
 
+export const getNavigationTargetNoteForWikiLink = async ({
+  data,
+  notes,
+}: {
+  data: NoteViewMessage["data"];
+  notes: NotePropsDict;
+}) => {
+  // wiki links will have the following format
+  //
+  // with `prettyLinks` set to false
+  //    with anchor: vscode-webview://4e98b9cf-41d8-49eb-b458-fcfda32c6c01/foo.html'
+  //
+  // with `prettyLinks` set to true
+  //    with anchor: vscode-webview://4e98b9cf-41d8-49eb-b458-fcfda32c6c01/foo'
+  //    without anchor: vscode-webview://4e98b9cf-41d8-49eb-b458-fcfda32c6c01/foo#foobar'
+  //
+  // And when the target of the link is a note in different vault without specifying
+  // the vault explicitly the href is going to have the file name of the node in place of the id:
+  // Example of href note inf different vault without vault specified:
+  // vscode-webview://25d7783e-df29-479c-9838-386c17dbf9b6/dendron.ref.links.target-different-vault
+  //
+  if (!data.href) {
+    throw ErrorFactory.createInvalidStateError({
+      message: `href is missing from data: '${ErrorFactory.safeStringify(
+        data
+      )}'`,
+    });
+  }
+
+  const noteId = extractNoteIdFromHref(data);
+  if (!noteId) {
+    throw ErrorFactory.createInvalidStateError({
+      message: `Failed to extract noteId from '${ErrorFactory.safeStringify(
+        data
+      )}'`,
+    });
+  }
+
+  const anchor = extractHeaderAnchorIfExists(data.href);
+  let note: NoteProps | undefined = notes[noteId];
+
+  if (note === undefined) {
+    // If we could not find the note by the extracted id (when the note is within the same
+    // vault we should have been able to find the note by id) there is a good chance that the name
+    // of the note was in place of the id in the HREF (as in case of navigating to a note
+    // in a different vault without explicit vault specification). Hence we will attempt
+    // to find the note by file name.
+    const candidates = NoteUtils.getNotesByFname({ fname: noteId, notes });
+
+    if (candidates.length === 1) {
+      note = candidates[0];
+    } else if (candidates.length > 1) {
+      // In case of multiple candidates matching the note we will grab the one with oldest (smallest)
+      // create date to make the navigation decently predictable (the predictability
+      // can be thrown off if a new vault is added to the workspace with matching name and older note).
+      note = candidates.reduce((n1, n2) => (n1.created < n2.created ? n1 : n2));
```

Comment:
its not intuitive to me that the older timestamp should be the valid note (eg. you could also add an existing vault with a note that was created with an earlier timestamp). as an alternative, what if we show the user a quickpick with the possible options? if you look at https://github.com/dendronhq/dendron/pull/1796, @hikchoi is adding the ability to create a lookup with only a few notes selected, seems like it would make a good use case for this

Reply:
Added quick pick choice if there are multiple notes in other vaults.